destination_test.go

```diff
@@ -1,15 +1,378 @@
+
+	written, err := dest.Write(ctx, []opencdc.Record{rec})
+	is.NoErr(err)
+	is.Equal(written, 1)
+
+	assertFileWrittenAndUnique(ctx, t, file, vectorStoreID)
+}
+
+func TestWrite_MultipleWritesWithSameKeyDontDuplicate(t *testing.T) {
+	ctx := testContext(t)
+	is := is.New(t)
+	var err error
+
+	vectorStoreID := createTestVectorStore(ctx, t)
+
+	dest := testDestination(ctx, t, vectorStoreID)
+
+	file1, file2 := newTestFile(), newTestFile()
+	file2.name = file1.name
+
+	written, err := dest.Write(ctx, []opencdc.Record{file1.recSnapshot(), file2.recCreate()})
+	is.NoErr(err)
+	is.Equal(written, 2)
+
+	assertFileWrittenAndUnique(ctx, t, file2, vectorStoreID)
+}
+
+func TestWrite_Snapshot(t *testing.T) {
+	ctx := testContext(t)
+	is := is.New(t)
+	var err error
+
+	vectorStoreID := createTestVectorStore(ctx, t)
+
+	dest := testDestination(ctx, t, vectorStoreID)
+
+	file := newTestFile()
+	rec := file.recSnapshot()
+
+	written, err := dest.Write(ctx, []opencdc.Record{rec})
+	is.NoErr(err)
+	is.Equal(written, 1)
+
+	assertFileWrittenAndUnique(ctx, t, file, vectorStoreID)
+}
+
+func TestWrite_Update(t *testing.T) {
+	ctx := testContext(t)
+	is := is.New(t)
+	var err error
+
+	vectorStoreID := createTestVectorStore(ctx, t)
+
+	dest := testDestination(ctx, t, vectorStoreID)
+
+	file := newTestFile()
+	rec := file.recSnapshot()
+
+	written, err := dest.Write(ctx, []opencdc.Record{rec})
+	is.NoErr(err)
+	is.Equal(written, 1)
+
+	file.contents = "updated contents"
+	rec = file.recUpdate()
+
+	written, err = dest.Write(ctx, []opencdc.Record{rec})
+	is.NoErr(err)
+	is.Equal(written, 1)
+
+	assertFileWrittenAndUnique(ctx, t, file, vectorStoreID)
+}
+
+func TestWrite_Delete(t *testing.T) {
+	ctx := testContext(t)
+	is := is.New(t)
+	var err error
+
+	vectorStoreID := createTestVectorStore(ctx, t)
+
+	dest := testDestination(ctx, t, vectorStoreID)
+
+	file := newTestFile()
+	rec := file.recSnapshot()
+
+	_, err = dest.Write(ctx, []opencdc.Record{rec})
+	is.NoErr(err)
+
+	rec = opencdc.Record{
+		Operation: opencdc.OperationDelete,
+		Key:       opencdc.RawData(file.name),
+	}
+
+	written, err := dest.Write(ctx, []opencdc.Record{rec})
+	is.NoErr(err)
+	is.Equal(written, 1)
+
+	assertDeletedFile(ctx, t, file.name)
+}
+
+func TestMain(m *testing.M) {
+	m.Run()
+
+	// it's much easier to clean up all test data after all tests have run, so
+	// that we don't worry of left over test data on each test.
+	// Assuming each test creates unique data, ofc.
+	cleanupAllTestData()
+}
+
+func testDestination(ctx context.Context, t *testing.T, vectorStoreID string) sdk.Destination {
+	t.Helper()
+	is := is.New(t)
+	dest := NewDestination()
+
+	err := dest.Configure(ctx, config.Config{
+		DestinationConfigApiKey:        getOpenAIApiKey(t),
+		DestinationConfigVectorStoreId: vectorStoreID,
+	})
+	is.NoErr(err)
+
+	err = dest.Open(ctx)
+	is.NoErr(err)
+
+	t.Cleanup(func() { is.NoErr(dest.Teardown(ctx)) })
+
+	return dest
+}
+
+func testClient(t *testing.T) *openai.Client {
```

Comment:
This utility method could be useful for maybe a source destination connector. I wonder if we should move it to another utils pkg or wait until we need it there.

Reply:
I generally agree with this pattern, but in this case I doubt we will ever need a source connector for openai vector stores.
I feel it's more straight forward to just put it inside `destination_test.go` with the other utilities.